src/Types/BigIntType.php

```diff
@@ -47,15 +46,19 @@ public function convertToPHPValue(mixed $value, AbstractPlatform $platform): int
             return $value;
         }
 
-        if ($value > PHP_INT_MIN && $value < PHP_INT_MAX) {
-            return (int) $value;
-        }
-
         assert(
             is_string($value),
             'DBAL assumes values outside of the integer range to be returned as string by the database driver.',
         );
 
+        if (str_starts_with($value, '+') || $value === '-0') {
```

Comment:
You do realize what I'm about to ask, don't you. 🙃

Reply:
I have removed even this, although I have done it only for you, my proposal is this PR without the last 3 commits as IMO the type should accept well formed integer string. All robust libs do that and in my company we use DBAL Types to cast data even from APIs for example. If you however persist on minimalistic impl., I am ok to give my code away even if I am not happy with it.

Reply:
I'm going to answer this in the main thread.